benchmarks/CallDepth.bench.cpp

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2022, Alibaba Group Holding Limited;
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "CallDepth.bench.h"
+#include <async_simple/coro/Collect.h>
+#include <async_simple/coro/Lazy.h>
+#include <async_simple/coro/SyncAwait.h>
+#include <async_simple/executors/SimpleExecutor.h>
+#include <async_simple/uthread/Async.h>
+#include <async_simple/uthread/Collect.h>
+#include <async_simple/uthread/Uthread.h>
+#include "ReadFileUtil.hpp"
+using namespace async_simple;
+using namespace async_simple::coro;
+using namespace async_simple::executors;
+using namespace async_simple::uthread;
+namespace fs = std::filesystem;
+static int task_num = 10;
+struct Service {
+    void run() { service1(); }
+
+    void service1() {
```

Comment:
we already has call depth test case in https://github.com/alibaba/async_simple/blob/main/async_simple/coro/test/LazyTest.cpp#L1307, maybe you can move to benchmark module. And add the other perf test like coroutine switch.